### ISSUE TITLE
How to Go to Next page ? How to retrieve next 20 Emails also what about attachments?

### ISSUE THREAD

**Issue by dhruv850** [KhalidLam/Gmail-App-React#2]
I think that is not issue but I am currently working on that I hope you can help me out there.

**Comment by KhalidLam** [comment]
Yes i would love to help you, this was just a side-project to learn React, but i'm planing to add more features in the future

You can send another messages.list request to get the next 20 emails but you should add pageToken to request parameters
![{167B2DAB-3CF1-4D57-B381-1D8FE7250241} png](https://user-images.githubusercontent.com/17090194/86675626-bb61e500-bff1-11ea-9693-05a595e18a99.jpg)

I've no idea how to add attachments right now  😁

**Comment by dhruv850** [comment]
Hey KhalidLam/CubeMail ,

I Tried to do this but somehow I am not able to do so. If you can please
show me I would really appreciate it, and I really love your work.

Thanks in advance

Dhruv Patel

**Comment by KhalidLam** [comment]
Okay I will submit new commit with function to get next page emails just to show you how, but it's just a temporary code😃

**Comment by dhruv850** [comment]
Thanks man , much appreciated. Let me know when you have done that.

Dhruv Patel

**Comment by KhalidLam** [comment]
Its already done 😁 check commits and let me know if you need help

**Comment by dhruv850** [comment]
Hey I Have seen the commits, but still it is not loading more emails when I
scroll down. Are you able to load more Emails?

**Comment by dhruv850** [comment]
I see what your added function do is it loads another 20 Emails but it is
not loading more after that. I am Just beginner into React and Node but I
look forward to include fully functional application including attachments
and everything. The function should be called recursively, let me know if
you are updating the code again. Thanks.

**Comment by KhalidLam** [comment]
No, I didn't display the new emails, the function only send a request to get new emails and display it in console, I'm planning to add an Infinite scroll if you want to display the new emails too you should uncomment this line in getNextPageMessages method
// setMessages((messages) => [...messages, response.result]);
 I'll fix this issue in the next few days don't worry 👌

**Comment by dhruv850** [comment]
Yes I managed to get next 20 emails by uncommenting this line , but I was
talking about infinite scroll until the end. As I have told you before I am
complete learner to this but I have found a great repo from github which
can help you. I am sharing you the link for it, It also have methods for
displaying attachments and even downloading it. I am sharing you link
https://github.com/kolya-t/Gmail-client .
navigate to "Actions" and "Message Methods" for this part. This repo also
has Infinite loop section but it is using redux so I don't have more idea.
But it can help you really. I appreciate your help I am too working on this
two parts. Please see the shared repo.

**Comment by dhruv850** [comment]
Have you got any success ?

**Comment by dhruv850** [comment]
Hey KhalidLam/CubeMail Have you got any success with that ? How to get
infinite scrolling ?

**Comment by dhruv850** [comment]
[image: image.png]

I was trying to get the attachments. somehow I am not able to implement it.
I really needs to finish this project XD.

**Comment by KhalidLam** [comment]
I'm working on other projects at the time so I cant implement it right now 😢

**Comment by dhruv850** [comment]
Come on man , It will be a piece of cake for you. Hoping for your help.

**Comment by dhruv850** [comment]
I have done the attachment part. Now I am able to get and download the
attachments.

Now I will work on send email with attachments.

Dhruv Patel

**Comment by KhalidLam** [comment]
Nice work, I'm working now on the infinite scrolling
I'll let you know when it's done

**Comment by dhruv850** [comment]
Hey , I have also implemented search functionality there, I have also
implemented reply with attachments and forward with adding the attachments.

Dhruv Patel

**Comment by KhalidLam** [comment]
Did you push the code on your Github account?  
I want to see the code, or you can make a pull request if you want

**Comment by dhruv850** [comment]
Hey KhalidLam/CubeMail,
I was not able to make pull request I think due to permission error. But
for you I have added my code to
https://github.com/dhruv850/Gmail-Client-React this repository.
1. Send email with attachments and get email with attachments including
downloading it working correctly
2. Added function to search the mail works correctly
3. Reply to email work correctly
4. To forward email I am able to pass the message but we need to manually
add the attachments. When we forward the email it does not forward
attachments it self but we can manually add it
5. I have added load more button but there is room for improvement there.

let me know if you have any questions, and I totally thank you for creating
the base.

**Comment by KhalidLam** [comment]
I
